streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1088,7 +1094,11 @@ private long computeLatency() {
      */
     public void shutdown() {
         log.info("Informed to shut down");
-        setState(State.PENDING_SHUTDOWN);
+        State oldState = setState(State.PENDING_SHUTDOWN);
+        if (oldState == State.CREATED) {
+            // Start so that we shutdown on the thread
+            this.start();
```

Comment:
Why don't we call `completeShutdown(true)` directly instead via the redirect to `run()` ?

Reply:
It seemed simpler to have all the shutdown stuff handled in the streams thread. I don't have a strong preference either way though.

Reply:
This lgtm.